File: Python/recursive_sequence.py
import sys
sys.setrecursionlimit(100000000)

n = int(input())


def recursive(n):
    if n == 0:
        return 5

    recursive_result = recursive(n - 1)
    if n % 2 == 0:
        return recursive_result - 21
    else:
        return recursive_result * recursive_result


print(recursive(n))

File: C++/recursive_sequence.cpp
#include <iostream>
#include <bits/stdc++.h>
using namespace std;

int recursive(int x){
    if(x == 0) return 5;

    int recursive_result = recursive(x - 1);

    if(x % 2 == 0) return recursive_result - 21;
    else return recursive_result * recursive_result;
};
int main(){
    int x ;
    cin >> x;
    cout << recursive(x);
    return 0;
}